src/navigation/rootNavigation.js

```diff
@@ -85,54 +84,52 @@ const App = ( ): React.Node => {
-                    component={ProjectsStackNavigation}
-                    options={hideHeader}
-                  />
-                  <Drawer.Screen name="settings" component={Settings} options={hideHeader} />
-                  <Drawer.Screen name="following (dashboard)" component={PlaceholderComponent} />
-                  <Drawer.Screen
-                    name="about"
-                    component={About}
-                  />
-                  <Drawer.Screen name="help/tutorials" component={PlaceholderComponent} />
-                  <Drawer.Screen name="login" component={MortalLogin} options={hideHeader} />
-                  <Drawer.Screen
-                    name="explore stack"
-                    component={ExploreStackNavigator}
-                    options={hideHeader}
-                  />
-                  <Drawer.Screen
-                    name="network"
-                    component={NetworkLogging}
-                  />
-                </Drawer.Navigator>
-              </ObsEditProvider>
-            </PhotoGalleryProvider>
+            <ObsEditProvider>
+              <Drawer.Navigator
+                screenOptions={screenOptions}
+                name="Drawer"
+                drawerContent={drawerRenderer}
+              >
+                <Drawer.Screen
+                  name="observations"
+                  component={ObservationsStackNavigation}
+                  options={hideHeader}
+                />
+                <Drawer.Screen
+                  name="notifications"
+                  component={NotificationsStackNavigation}
+                  options={hideHeader}
+                />
+                <Drawer.Screen
+                  name="identify"
+                  component={IdentifyStackNavigation}
+                  options={hideHeader}
+                />
+                <Drawer.Screen name="search" component={Search} />
+                <Drawer.Screen
+                  name="projects"
+                  component={ProjectsStackNavigation}
+                  options={hideHeader}
+                />
+                <Drawer.Screen name="settings" component={Settings} options={hideHeader} />
+                <Drawer.Screen name="following (dashboard)" component={PlaceholderComponent} />
+                <Drawer.Screen
+                  name="about"
+                  component={About}
+                />
+                <Drawer.Screen name="help/tutorials" component={PlaceholderComponent} />
+                <Drawer.Screen name="login" component={MortalLogin} options={hideHeader} />
+                <Drawer.Screen
+                  name="explore stack"
+                  component={ExploreStackNavigator}
+                  options={hideHeader}
+                />
+                <Drawer.Screen
+                  name="network"
+                  component={NetworkLogging}
+                />
+              </Drawer.Navigator>
+            </ObsEditProvider>
```

Comment:
So remind me, is one of the problems here is that we don't want to expose photo gallery state to the entire app, but we can't wrap a provider around a subset of screens? Like this won't work?

```javascript
<Stack.Navigator>
  <Stack.Screen
    name="ObsList"
    component={ObsList}
  />
  <PhotoGalleryProvider>
    <Stack.Screen
      name="PhotoGallery"
      component={PhotoGalleryWithPermission}
    />
    <Stack.Screen
      name="GroupPhotos"
      component={GroupPhotos}
    />
  </PhotoGalleryProvider>
</Stack.Navigator>
```

Reply:
Yes, as far as I know, we can't define a provider around a subset of screens. I think if we were able to break out certain screens into their own stack navigator, then we could wrap a provider around just those screens -- but that's probably not desirable for the photo gallery since we need to be able to navigate back from ObsEdit.